plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -211,11 +288,132 @@ public FormValidation doTestBrokerConnection(@QueryParameter("brokerURL") final
+
+        try {
+            String serverUrl = new URIBuilder()
+                    .setHost(serverIp)
+                    .setPort(Integer.parseInt(serverPort))
+                    .toString();
+            KubernetesClient client = new KubernetesFactoryAdapter(serverUrl, namespace,
+                    Util.fixEmpty(serverCertificate), Util.fixEmpty(credentialsId), skipTlsVerify
+            ).createClient();
+            Class clazz = GlobalKafkaConfiguration.class;
+            client.load(clazz.getResourceAsStream("kubernetes/zookeeper.yaml")).createOrReplace();
+            LOGGER.info("Starting Zookeeper");
+            client.load(clazz.getResourceAsStream("kubernetes/kafka-service.yaml")).createOrReplace();
+            LOGGER.info("Starting Kafka Services");
+            Integer zookeeperPort = KubernetesUtils.getFirstNodePortByServiceName(client, "zookeeper-svc");
+            Integer kafkaPort = KubernetesUtils.getFirstNodePortByServiceName(client, "kafka-svc");
+
+            // Set Kafka advertised.listeners property
+            List<HasMetadata> kafkaStatefulSetResources =
+                    client.load(clazz.getResourceAsStream("kubernetes/kafka-statefulset.yaml")).get();
+            StatefulSet kafkaStatefulSet = (StatefulSet) kafkaStatefulSetResources.
+                    stream().
+                    filter(res -> res.getKind().equals("StatefulSet") && res.getMetadata().getName().equals("kafka")).
+                    findFirst().
+                    orElse(null);
+            if (kafkaStatefulSet == null)
+                throw new RemotingKafkaConfigurationException("Couldn't find StatefulSet named kafka in YAML configuration");
+            Container kafkaContainer = kafkaStatefulSet.
+                    getSpec().
+                    getTemplate().
+                    getSpec().
+                    getContainers().
+                    stream().
+                    filter(con -> con.getName().equals("kafka")).
+                    findFirst().
+                    orElse(null);
+            if (kafkaContainer == null)
+                throw new RemotingKafkaConfigurationException("Couldn't find Container named kafka in template specification");
+            kafkaContainer.getEnv().add(new EnvVar(
+                    "KAFKA_ADVERTISED_LISTENERS",
+                    String.format("EXTERNAL://%s:%s", serverIp, kafkaPort),
+                    null
+            ));
+            client.resourceList(kafkaStatefulSetResources).createOrReplace();
+            LOGGER.info("Starting Kafka StatefulSet");
+
+            // Probe for Kafka readiness
+            while (true) {
+                try {
+                    testConnection(serverIp, kafkaPort);
+                    break;
+                } catch (IOException e) {
+                    LOGGER.info(String.format("Waiting for Kafka connection at %s:%s", serverIp, kafkaPort));
+                }
+                Thread.sleep(2000);
```

Comment:
Used TimeUnit for readability. Also any particular reason for 2 seconds?

Reply:
@afalko there isn't, I just thought that 2 seconds would be a reasonable time period